### PR TITLE
Add one-off sql management

### LINKED ISSUE #166: Correct proj extension values <and schema> in pgstac
# What
The batch loading step of our ingest system caused proj values to be stored as float instead of int. This is acceptable for the older version of the proj extension (0.0.1) but requires a work around for users of the VEDA stac-api. 
1. `proj` values should be stored as int
2. `proj` extension should be upgraded to 0.1.0 _if_ it is decided to upgrade the extension for newly created records.

For forward ingests, the values will be fixed in an [upgrade to the ingest API](https://github.com/NASA-IMPACT/veda-architecture/issues/215) and the stac extension for new records is addressed in [this pipelines tasks issue](https://github.com/NASA-IMPACT/veda-pipeline-tasks/issues/9). When the ingest systems are updated, this issue is to apply a backwards correction to all of the currently ingested records. If this can be executed safely in the database, engage @bitner for a review, if the backward correction cannot be safely executed, open issues to re-ingest impacted records.

## Bug reported in pipelines
https://github.com/NASA-IMPACT/veda-data-pipelines/issues/313

# AC
- [x] `proj` values should be stored as int
- [ ] `proj` extension should be upgraded to 0.1.0 _if_ it is decided to upgrade the extension for newly created records.

### PR DESCRIPTION
**Pull request by moradology**
1. Created a base class `PostgreSQLCommandRunner` that abstracts the execution of SQL commands against a PostgreSQL database.

2. Introduced `ProjExtensionFloatIntFix`, a subclass of `PostgreSQLCommandRunner`, to convert floating point values to integers for the projection extension of the STAC in PostgreSQL.

3. Implemented the `DeleteNullStacExtensions `class, a subclass of PostgreSQLCommandRunner, to delete the stac_extensions field from collections that currently have it marked as null in a PostgreSQL database. 

4. Added Markdown documentation for these classes and methods. 

Closes #166

### CODE REVIEW

**Comment by anayeaye** [comment]
This is so useful!
Does it make sense to include these database update tools in the main project [support scripts](https://github.com/NASA-IMPACT/veda-backend/tree/develop/support_scripts)? All that we have in the project support scripts is a CLI tool for rotating postgres and pgstac passwords which seems related enough to _consider_ consolidating. OR maybe just add a reference to the command runner documentation in the support scripts readme or the main [readme](https://github.com/NASA-IMPACT/veda-backend#support-scripts)? Also if you have any ideas about a better way to describe these veda backend operator tools, I would be very glad to rename 'support scripts'.

**Review by jsignell — COMMENTED** [review]
I really like the idea of this functionality! It feels surprising that this much boilerplate is needed to just execute a SQL command, but I like the class structure that you have set up. I just have a few minor comments

**Comment by moradology** [comment]
Yeah, it is possible that this is overkill but I'm erring on the side of reducing friction so there is more sugar here than you might normally encounter. Also, I'm hoping this can serve as documentation of prior efforts/a way to make solutions look similar to one another instead of invented out of whole cloth.

File: support_scripts/sql_commands/PostgreSQLCommandRunner.py
[the remark this answers is not in the document]
Comment (moradology) [reply support_scripts/sql_commands/PostgreSQLCommandRunner.py:1]: Symptoms of JVM-poisoning, I fear

**Review by slesaad — APPROVED** [review]
love it!

**Comment by anayeaye** [comment]
I still want to have a closer look at how the update is executed, but I'm working through some of the examples from the original bug report [here](https://github.com/NASA-IMPACT/veda-data-pipelines/issues/313) and things are looking good (no surprise 😀, just wanted to see).  

For example running this for dev and staging shows that the correction in dev worked and `proj:epsg` is now int:
```
item = pystac.Item.from_file("https://dev-stac.delta-backend.com/collections/nceo_africa_2017/items/AGB_map_2017v0m_COG")
print(ProjectionExtension.has_extension(item))  # False
print(item.properties["proj:epsg"]) 
```

**Review by anayeaye — APPROVED** [review]
Spot checked some of the updated records in the dev pgstac database and all look correct. Let's wait to execute in staging until after the 0.7.* migration issue is resolved just in case? The operation looks safe but less moving parts is good when resolving that partitioning issue.

**Pull request merged**